src/ScottPlot5/ScottPlot/Control/Backend.cs

```diff
@@ -9,107 +7,211 @@ namespace ScottPlot.Control
 
-        /// <summary>
-        /// This object is used to pair mouse interactions with plot actions.
-        /// The default behavior is left-click-drag pan and right-click drag-zoom,
-        /// but advanced users can use their own interaction system instead.
-        /// </summary>
-        public IInteractions Interactions;
+        private Plot Plot => Control.Plot;
 
-        private readonly KeyStates KeyStates = new();
+        private readonly IPlotControl Control;
 
-        private readonly MouseButtonStates MouseButtonStates = new();
+        private readonly KeyboardState Keyboard = new();
 
-        /// <summary>
-        /// Latest position of the mouse (in pixel units)
-        /// </summary>
-        public Pixel MousePosition { get; private set; } = new(float.NaN, float.NaN);
+        private readonly MouseState Mouse = new();
+
+        private Pixel MousePosition = new(float.NaN, float.NaN);
+
+        private double ZoomInFraction { get; set; } = 1.15;
+
+        private double ZoomOutFraction { get; set; } = 0.85;
+
+        private double PanFraction { get; set; } = 0.1;
 
         /// <summary>
-        /// Latest position of the mouse (in coordinate units)
+        /// Create a backend for a user control to manage interaction and event handling.
         /// </summary>
-        public Coordinates GetMouseCoordinates()
+        /// <param name="plotControl">The control whose plot is being controlled by this backend</param>
+        public Backend(IPlotControl control)
         {
-            return Plot.GetCoordinate(MousePosition);
+            Control = control;
         }
 
         /// <summary>
-        /// Create a backend for a user control to manage interaction and event handling.
+        /// Latest position of the mouse (in coordinate units)
         /// </summary>
-        /// <param name="plotControl">The control whose plot is being controlled by this backend</param>
-        public Backend(IPlotControl plotControl)
+        public virtual Coordinates GetMouseCoordinates()
         {
-            Plot = plotControl.Plot;
-            Interactions = new StandardInteractions(plotControl);
+            return Plot.GetCoordinate(MousePosition);
         }
 
-        public void MouseDown(Pixel position, MouseButton button)
+        public virtual void MouseDown(Pixel position, MouseButton button)
         {
-            MouseButtonStates.Down(position, button, Plot.GetAxisLimits());
-            Interactions.MouseDown(position, button, KeyStates.PressedKeys);
+            Mouse.Down(position, button, Plot.GetAxisLimits());
         }
 
-        public void MouseUp(Pixel position, MouseButton button)
+        public virtual void MouseUp(Pixel position, MouseButton button)
         {
-            bool drag = MouseButtonStates.IsDragging(position);
+            bool drag = Mouse.IsDragging(position);
+
+            // TODO: when should the rectangle be cleared?
 
             if (drag)
```

Comment:
I think we should keep `MouseDragEnd` as a separate function, that way we don't mix the interpretation of events (e.g. was this a drag, was this key held, etc) with acting upon them. If we discover (or introduce) a problem we have a lot more places to check when it can read state wherever it wants rather than in a single function call. Especially if we start supporting more complicated behaviour.

I'd also note that it would be much easier for users to customize behaviour if there's a distinct `MouseDragEnd` event to override, rather than them having to to override (and clobber) `MouseUp` and reimplement drag-detection. Conceptually I think `MouseUp` and `MouseDragEnd` are very different events which deserve distinct handlers, even if every `MouseDragEnd` has a `MouseUp`. Similar to how we have `MouseMove` and `MouseDrag`